lib/profiling/lfric/profile_psy_data_mod.F90

```diff
@@ -0,0 +1,119 @@
+!
+! THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+! AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+! IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+! DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+! FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+! DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+! SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+! CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+! OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+! OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+! -----------------------------------------------------------------------------
+! Authors J. Henrichs, Bureau of Meteorology
+
+
+!> An implemention of the PSyData API for profiling, which wraps the use
+!> of the LFRic timer code.
+
+module profile_psy_data_mod
+  type :: profile_PSyDataType
+     character(:), allocatable :: name
+     logical                   :: initialised = .false.
+
+  contains
+      ! The profiling API uses only the two following calls:
+      procedure :: PreStart, PostEnd
+  end type profile_PSyDataType
+
+
+contains
+  ! ---------------------------------------------------------------------------
+  !> The initialisation subroutine. It is not called directly from
+  !! any PSyclone created code, so a call to profile_PSyDataInit must be
+  !! inserted manually by the developer.
+  !!
+  subroutine profile_PSyDataInit()
+    use timer_mod, only: init_timer
+
+    implicit none
+    call init_timer()
+
+  end subroutine profile_PSyDataInit
+
+  ! ---------------------------------------------------------------------------
+  !> Starts a profiling area. The module and region name can be used to create
+  !! a unique name for each region.
+  !! Parameters:
+  !! @param[inout] this This PSyData instance.
+  !! @param[in] module_name Name of the module in which the region is
+  !! @param[in] region_name Name of the region (could be name of an invoke, or
+  !!            subroutine name).
+  !! @param[in] num_pre_vars The number of variables that are declared and
+  !!            written before the instrumented region, which is always 0 for
+  !!            a profiling wrapper, and not  used.
+  !! @param[in] num_post_vars The number of variables that are also declared
+  !!            before an instrumented region of code, but are written after
+  !!            this region, which is always 0 for a profiling wrapper, and
+  !!            not used.
+  subroutine PreStart(this, module_name, region_name, num_pre_vars, &
+                      num_post_vars)
+    use timer_mod, only: timer
+    implicit none
+
+    class(profile_PSyDataType), intent(inout), target :: this
+    character*(*) :: module_name, region_name
```

Comment:
Character* is obsolete so the Intel complier errors out when compiling this, changed it to just compiler(*) and everything worked fine.

Reply:
@hiker would you mind addressing this and then I can proceed to merge? Thanks :-)

Reply:
*grin* I show my age there :) Which compiler version are you using? I had verified this with intel 17 and intel 21 (and did just again), and both are happy with this. I just want to be able test it (since I've used character*(*) in a few places, I'd prefer to fix them all at once).

Reply:
I just did a rerun myself (we use 17 as well) and i can't reproduce the error now, running the test-suite again to see if it was a specific app that flagged it up. I have checked and it was made obsolete as of F95 so at least i'm not imagining things.

Reply:
Ok so gets flagged when compiling everything but the infrastructure unit tests, perhaps we have a flag on that you don't? our flags are set in infrastructure/build/fortran/ifort.mk if you want to have a look.

Reply:
``-stand f08`` flags it, but it flags even more stuff - anyway, it's merged. Thanks